batch/plugins/org.jboss.tools.batch.ui/src/org/jboss/tools/batch/ui/hyperlink/BatchPropertyDialog.java

```diff
@@ -0,0 +1,460 @@
+		tree.setAutoExpandLevel(TreeViewer.ALL_LEVELS);
+		GridData g = new GridData(GridData.FILL_BOTH);
+		
+		tree.getControl().setLayoutData(g);
+		tree.setContentProvider(new TreeContent());
+		tree.setLabelProvider(new LabelProvider());
+		tree.setInput(batchProperty);
+		patternFilter = new ReferencePatternFilter();
+		tree.setFilters(new ViewerFilter[]{patternFilter});
+		tree.addOpenListener(new IOpenListener() {			
+			@Override
+			public void open(OpenEvent event) {
+				ISelection s = event.getSelection();
+				if(!s.isEmpty() && s instanceof IStructuredSelection) {
+					Object o = ((IStructuredSelection)s).getFirstElement();
+					if(o instanceof NodePathTextSourceReference) {
+						openElement((NodePathTextSourceReference) o);
+						close();
+					}
+				}
+			}
+		});
+
+		tree.getTree().addMouseMoveListener(new MouseMoveListener() {
+			@Override
+			public void mouseMove(MouseEvent e) {
+				ViewerCell cell = tree.getCell(new Point(e.x, e.y));
+				if(cell != null) {
+					Widget w = cell.getItem();
+					if(w != null && w.getData() != null) {
+						tree.setSelection(new StructuredSelection(w.getData()));
+					}
+				}
+			}
+		});
+		tree.getTree().addMouseListener(new MouseAdapter() {
+			@Override
+			public void mouseUp(MouseEvent e) {
+				gotoSelectedElement();
+			}
+		});
+		
+
+		tree.refresh();
+	}
+
+	protected void gotoSelectedElement() {
+		ISelection s = tree.getSelection();
+		if(!s.isEmpty() && s instanceof IStructuredSelection) {
+			Object o = ((IStructuredSelection)s).getFirstElement();
+			if(o instanceof NodePathTextSourceReference) {
+				openElement((NodePathTextSourceReference) o);
+				close();
+			}
+		}
+	}
+
+	
+	protected Text getFilterText() {
+		return fFilterText;
+	}
+
+	protected Text createFilterText(Composite parent) {
+		fFilterText= new Text(parent, SWT.NONE);
+		Dialog.applyDialogFont(fFilterText);
+
+		GridData data= new GridData(GridData.FILL_HORIZONTAL);
+		data.horizontalAlignment= GridData.FILL;
+		data.verticalAlignment= GridData.CENTER;
+		fFilterText.setLayoutData(data);
+		
+
+		fFilterText.addKeyListener(new KeyListener() {
+			@Override
+			public void keyPressed(KeyEvent e) {
+				if (e.keyCode == 0x0D) // return
+					gotoSelectedElement();
+				if (e.keyCode == SWT.ARROW_DOWN)
+					tree.getTree().setFocus();
+				if (e.keyCode == SWT.ARROW_UP)
+					tree.getTree().setFocus();
+				if (e.character == 0x1B) // ESC
+					close();
+			}
+			@Override
+			public void keyReleased(KeyEvent e) {
+				// do nothing
+			}
+		});
+		return fFilterText;
+	}
+
+	private void installFilter() {
+		fFilterText.setText(""); //$NON-NLS-1$
+
+		fFilterText.addModifyListener(new ModifyListener() {
+			@Override
+			public void modifyText(ModifyEvent e) {
+				//textFilter.filter();
+				patternFilter.setPattern(fFilterText.getText());
+				tree.refresh();
+				tree.expandAll();
+			}
+		});
+	}
+
+	protected String getId() {
```

Comment:
This method was declared in AssignableBeansDialog to override

protected IDialogSettings getDialogSettings() {}

which in PopupDialog returns null. I recommend following the pattern and override this method here, too. When settings are available, even if we have nothing specific to keep, popup dialog has a couple of useful things to store in it.